lib/widgets/message_list.dart

```diff
@@ -1018,11 +1003,25 @@ class StreamMessageRecipientHeader extends StatelessWidget {
   const StreamMessageRecipientHeader({
     super.key,
     required this.message,
-    required this.showStream,
+    required this.narrow,
   });
 
   final StreamMessage message;
-  final bool showStream;
+  final Narrow narrow;
+
+  static bool _containsDifferentChannels(Narrow narrow) {
```

Comment:
nit:

    msglist [nfc]: Move _containsDifferentChannels to StreamMessageRecipientHeader
    
    Move the _containsDifferentChannels method to StreamMessageRecipientHeader where
    it's used, and pass narrow directly instead of a pre-computed showStream boolean.

This body is too wide, at 82 columns. Prose should wrap at around 68 columns, or 70 at max. See our Git style guide:
https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#formatting-guidelines